codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/protocol/ServerProtocolTestGenerator.kt

```diff
@@ -93,7 +93,7 @@ class ServerProtocolTestGenerator(
             val testCases = error.getTrait<HttpResponseTestsTrait>()?.testCases.orEmpty()
             testCases.map { TestCase.ResponseTest(it, error) }
         }
-        val allTests: List<TestCase> = (requestTests + responseTests + errorTests).filterMatching()
```

Comment:
Note that I have removed generation of [HTTP protocol compliance response tests](https://awslabs.github.io/smithy/1.0/spec/http-protocol-compliance-tests.html?highlight=httpresponsetest#httpresponsetests). The generated tests were of the form:

```rust
/// Test ID: RegisterServiceResponseTest
#[tokio::test]
async fn register_service_response_test_response() {
    let expected_output = crate::output::RegisterServiceOutput::builder()
        .set_id(Some("1".to_string()))
        .set_name(Some("TestService".to_string()))
        .build();
    use aws_smithy_http::response::ParseStrictResponse;
    use aws_smithy_http_server::response::SerializeHttpResponse;
    let op = crate::operation::RegisterService::new();
    let http_response = op
        .serialize(&expected_output)
        .expect("unable to serialize response body");
    let parsed = op.parse(&http_response);
    let parsed = parsed.expect("unable to parse response");
    assert_eq!(parsed.id, expected_output.id, "Unexpected value for `id`");
    assert_eq!(
        parsed.name, expected_output.name,
        "Unexpected value for `name`"
    );
}
```

Of course these tests now fail, because I've removed the `parse()` HTTP responses method on operations, since the server doesn't need to parse responses, only requests.

This led me to read the Smithy spec and the purpose of these tests, and I think we are doing them wrong. I think what we should be doing is instantiating an operation output using the `params` in the model, serializing that into an HTTP response, and directly inspecting the status code, headers, body etc. of that response and asserting that they match whatever was written in the model.

Is my understanding correct? If so, I can readd the tests in another PR but implementing them like I described in the previous paragraph.

Reply:
Your understanding is correct. The protocol tests have to be changed to 1) adhere to the new public API 2) do the right thing, which currently they are not doing.

Reply:
https://github.com/awslabs/smithy-rs/pull/840